app/assets/javascripts/likes.js

```diff
@@ -1,2 +1,17 @@
-// Place all the behaviors and hooks related to the matching controller here.
-// All this logic will automatically be available in application.js.
+function ready(fn) {
+  if (document.readyState != 'loading'){
+    fn();
+  } else {
+    document.addEventListener('DOMContentLoaded', fn);
+  }
+}
```

Comment:
I'd extract the `ready` function to another file and use rails require comments to require it first in `application.js`. That way we won't have it repeated in `avatar.js` and `likes.js`.

Bear in mind you'll need to test the avatar fnality too to make sure it works!